### PR TITLE
curl package upgraded to 7.81.0

### PR DESCRIPTION
**Pull request by ghost**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
To fix vulnerabilities reported by WhiteSource

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
LGTM.

But the proper solution will be to remove `curl`.
It is used only to send data to `sentry`. We should replace it to something minimal or remove altogether.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
@Meena-Renganathan  It does not build on Mac OS X,
see the "special build check".

**Comment by alexey-milovidov** [comment]
@Meena-Renganathan Let's fix the build or close this.

**Comment by Meena-Renganathan** [comment]
Sorry, I'm working on it. I will make the changes to fix the build error soon.

**Comment by Meena-Renganathan** [comment]
Hi @alexey-milovidov, I've tried to reproduce the error by building curl (with openssl option) in my local MacOS but it's getting built successfully. Here is the details of my MacOS:

macOS Monterey
Clang Version 13.

 CC    libcurl_la-hostip.lo
 CC    libcurl_la-hostip4.lo
 CC    libcurl_la-hostip6.lo
 CC    libcurl_la-hostsyn.lo

Could you help how it can take it up further to get this fixed?

**Comment by alexey-milovidov** [comment]
Let's check the build log: https://s3.amazonaws.com/clickhouse-builds/35130/d19090fdaf89162ba5527f7f376394a6cdd4081f/binary_darwin/build_log.log

Maybe it is related to some defines.
We can investigate it by running exactly the same compiler command.

Also another suggestion: maybe remove curl instead of updating it?

**Comment by Meena-Renganathan** [comment]
Thank you @alexey-milovidov. But, while trying upgrade the curl package to 7.81.0, observed the curl is getting referenced by sentry-native and azure modules. So, decided to upgrade the curl.

**Comment by Meena-Renganathan** [comment]
Thank you @alexey-milovidov. But, while trying upgrade the curl package to 7.81.0, observed the curl is getting referenced by sentry-native and azure modules. So, decided to upgrade the curl.

**Comment by alexey-milovidov** [comment]
Is it possible to remove curl from Azure?

**Comment by Meena-Renganathan** [comment]
It is not possible to directly remove the curl dependency from azure.

Azure module uses the  libcurl and WinHTTP libraries as HTTP stacks for communicating with Azure services over the network. When no specific configuration is set, the default transport adapter will be selected based on the OS. But custom transport adapter needs to be implemented to avoid the dependency on curl.

**Comment by alexey-milovidov** [comment]
Ok, then we need to fix this remaining build.

PS. There was similar problem in AWS library. They used curl and it lead to buffering all data in memory (I believe that it is not a downside of a curl, it is actually a good, very well tested library), and we replaced it by Poco. Maybe do similar with Azure?

**Comment by alexey-milovidov** [comment]
Still not fixed.

**Comment by Meena-Renganathan** [comment]
Hi @alexey-milovidov, yeah noticed the error and working on it to fix.

File: contrib/curl-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -153,6 +153,22 @@ set (SRCS
 
 add_library (_curl ${SRCS})
 
+set(CURL_LIBS "")
+if(APPLE)
+
+    find_library(SYSTEMCONFIGURATION_FRAMEWORK "SystemConfiguration")
```
Comment (alexey-milovidov) [inline contrib/curl-cmake/CMakeLists.txt:159]: We don't allow `find_library` in our cmake files.

Comment (Meena-Renganathan) [reply contrib/curl-cmake/CMakeLists.txt:159]: Hi @alexey-milovidov, any specific reason we are restricting the find_library in cmake files?

Comment (alexey-milovidov) [reply contrib/curl-cmake/CMakeLists.txt:159]: It is needed for hermetic builds - to avoid dependence on the environment.

Comment (alexey-milovidov) [reply contrib/curl-cmake/CMakeLists.txt:159]: If the new curl version started to need something from SystemConfiguration,
maybe better to chop it from the library?

**Comment by alexey-milovidov** [comment]
I did not check, but need to figure out where exactly these dependencies are needed? Is it possible to avoid them?

**Comment by alexey-milovidov** [comment]
Undefining this macro would help: https://github.com/curl/curl/blob/8a1fa3b364b9db52cf3e8d843ac1e749890985e6/lib/hostip.c#L71

**Comment by Meena-Renganathan** [comment]
Hi @alexey-milovidov,

To get better clarity, underlining the macro ENABLE_IPV6 doesn't allow to support IPV6 addressing format in ClickHouse. Is that my understanding, correct?

**Comment by alexey-milovidov** [comment]
No. We must support IPv6.

**Comment by alexey-milovidov** [comment]
`CURL_OSX_CALL_COPYPROXIES`

This.

I don't know what is "copy proxies" on OS X, and so we don't need it.

**Comment by alexey-milovidov** [comment]
Also please grep the code base for all instances of CURL_OSX_*.

**Comment by Meena-Renganathan** [comment]
Undefining this macro would require a code change in curl repo (curl/lib/curl_setup.h).

**Comment by alexey-milovidov** [comment]
Another option is to disable `curl`, `azure` and `sentry` on Mac OS, so the problem will not exist.

**Comment by alexey-milovidov** [comment]
Also getting rid of `curl` is a good option because we don't need it.
It can be done easily by implementing `Azure::Core::Http::HttpTransport`.

**Comment by Meena-Renganathan** [comment]
I'm planning to proceed with the option of disabling curl, azure and sentry on Mac OS.

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Comment by alexey-milovidov** [comment]
This is ready to be merged.
Need to merge with master one more time.

**Comment by alexey-milovidov** [comment]
Ok, let's try as is.

**Comment by azat** [comment]
>Let's check the build log: https://s3.amazonaws.com/clickhouse-builds/35130/d19090fdaf89162ba5527f7f376394a6cdd4081f/binary_darwin/build_log.log

Looks like the problem with `bool` type.

**Comment by alexey-milovidov** [comment]
ClickHouse build check (actions) — 22/22 builds are OK
ClickHouse special build check (actions) — 8/8 builds are OK

**Pull request merged**